pkg/controller/atlasdeployment/atlasdeployment_controller.go

```diff
@@ -477,3 +482,24 @@ func (r *AtlasDeploymentReconciler) removeDeletionFinalizer(context context.Cont
 }
 
 type deploymentHandlerFunc func(ctx *workflow.Context, project *mdbv1.AtlasProject, deployment *mdbv1.AtlasDeployment, req reconcile.Request) (workflow.Result, error)
+
+// Parse through tags and verfiy that all keys are unique. Return error on duplicate key.
+func uniqueKey(deploymentSpec *mdbv1.AtlasDeploymentSpec) (workflow.Result, error) {
```

Comment:
I think we do not need 2 results here. Error would suffice.

```suggestion
func uniqueKey(deploymentSpec *mdbv1.AtlasDeploymentSpec) error {
```

That will simplify the returns here and then on the calling spot you just handle the result as needed:

```go
	if err := uniqueKey(&deployment.Spec); err != nil {
		log.Errorw("failed to validate tags", "error", err)
		result := workflow.Terminate(workflow.Internal, err.Error())
		ctx.SetConditionFromResult(status.DeploymentReadyType, result)
		return result.ReconcileResult(), nil
	}
```

Not a big deal anyway, but the unit test will also be a bit easier.

Reply:
Noted, I'll push these changes with my next commit!

Reply:
Another option is to use just `workflow.Result`, as is can always wrap an error inside as needed (the error message).